src/context/toast/index.tsx

```diff
@@ -0,0 +1,157 @@
+  useState,
+} from "react";
+import styled from "@emotion/styled";
+import Toast, { Variant } from "@leafygreen-ui/toast";
+import {
+  TOAST_TIMEOUT,
+  mapToastToLeafyGreenVariant,
+  mapVariantToTitle,
+} from "constants/toast";
+import { zIndex } from "constants/tokens";
+import { DispatchToast, VisibleToast } from "types/toast";
+
+export interface DispatchToastContextState {
+  success: DispatchToast;
+  warning: DispatchToast;
+  error: DispatchToast;
+  info: DispatchToast;
+  progress: DispatchToast;
+  hide: () => void;
+}
+
+export const ToastDispatchContext =
+  createContext<DispatchToastContextState | null>(null);
+
+const useToastContext = (): DispatchToastContextState => {
+  const context = useContext(ToastDispatchContext);
+  if (context === null || context === undefined) {
+    throw new Error("useToastContext must be used within a ToastProvider");
+  }
+  return context;
+};
+
+const ToastProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [toastOpen, setToastOpen] = useState(false);
+  const [visibleToast, setVisibleToast] = useState<VisibleToast>({
+    variant: Variant.Note,
+    title: "",
+    message: "",
+    closable: true,
+    onClose: () => {},
+    shouldTimeout: true,
+  });
+
+  useEffect(() => {
+    if (!visibleToast.shouldTimeout) {
+      return;
+    }
+    const timeout = setTimeout(() => {
+      hideToast();
+    }, TOAST_TIMEOUT);
+    return () => clearTimeout(timeout);
+  });
+
+  const dispatchToast = useCallback(
+    (toast: VisibleToast) => {
+      setVisibleToast(toast);
+      setToastOpen(true);
+    },
+    [setVisibleToast, setToastOpen]
+  );
+
+  const hideToast = useCallback(() => {
+    setToastOpen(false);
+  }, [setToastOpen]);
+
+  const toastContext = useMemo(() => {
+    const defaultOptions = {
+      onClose: () => {},
+      shouldTimeout: true,
+      title: "",
+    };
+
+    return {
+      success: (message = "", closable = true, options = {}) =>
+        dispatchToast({
+          variant: mapToastToLeafyGreenVariant.success,
+          message,
+          closable,
+          ...defaultOptions,
+          ...options,
+        }),
+      warning: (message = "", closable = true, options = {}) =>
+        dispatchToast({
+          variant: mapToastToLeafyGreenVariant.warning,
+          message,
+          closable,
+          ...defaultOptions,
+          ...options,
+        }),
+      error: (message = "", closable = true, options = {}) =>
+        dispatchToast({
+          variant: mapToastToLeafyGreenVariant.error,
+          message,
+          closable,
+          ...defaultOptions,
+          ...options,
+        }),
+      info: (message = "", closable = true, options = {}) =>
+        dispatchToast({
+          variant: mapToastToLeafyGreenVariant.info,
+          message,
+          closable,
+          ...defaultOptions,
+          ...options,
+        }),
+      progress: (message = "", closable = true, options = {}) =>
```

Comment:
I think we should type the optional progress param

Reply:
i didnt even see that, that's a cool prop 🤯